emcc.py

```diff
@@ -1123,7 +1124,7 @@ def check(input_file):
       shared.Settings.EXPORTED_FUNCTIONS += ['___cxa_demangle']
       forced_stdlibs += ['libc++abi']
 
-    if not shared.Settings.ONLY_MY_CODE:
+    if not shared.Settings.ONLY_MY_CODE and not shared.Settings.MINIMAL_RUNTIME:
       # Always need malloc and free to be kept alive and exported, for internal use and other modules
       shared.Settings.EXPORTED_FUNCTIONS += ['_malloc', '_free']
```

Comment:
Unconditionally adding malloc and free should not occur, but it should only happen based on linking decisions dictating that they are needed.